### ISSUE TITLE
url slug a big problem

### ISSUE THREAD

**Issue by nestle49** [humhub/wiki#223]
/wiki/page/view?title=Ценности

We can't translit last of part url.
I try to change **source code** (!!!) module and discover, slug and title page are the same field in database. Whats'up, guys?

Also, I can't change or extend funcionality rich text editor wiki, for example to edit source html code.

**Comment by luke-** [comment]
What do you mean with "We can't translit last of part url." ?

**Comment by nestle49** [comment]
Non-English characters are not transliterated into English-language person-like urls.

For example, 'development department' in url looks like 'title = department + development', but I would like 'title = otdel-razrabotki'

I went into the sources and found that you are using the title field both as the slug / url of the page and to save the page. Although for saving it would be necessary to use id, and store slug / url as a separate field. It is very strange.

**Comment by aaukt** [comment]
Using the title as the URL slug is indeed not optimal. It poses too many restrictions on the title because the title has to be "url compatible".

In part @yurabakhtin spoke about it in this comment: https://github.com/humhub/wiki/issues/136#issuecomment-866646662

In general a slug could be created like this (JS pseudo-code):

```
title.toString()
.toLowerCase()                  // Convert the string to lowercase letters
.normalize('NFD')       // The normalize() method returns the Unicode Normalization Form of a given string.
.trim()                         // Remove whitespace from both sides of a string
.replace(/\s+/g, '-')           // Replace spaces with -
.replace(/[^\w\-]+/g, '')       // Remove all non-word chars
.replace(/\-\-+/g, '-');        // Replace multiple - with single -
```
Translation/conversion of special characters (äöüøá...) could/would be on top of that.

The slug could be an extra field in edit mode that gets generated automatically from the title in the frontend, but is still possible to overwrite manually.
